types/release_metadata.go

```diff
@@ -5,11 +5,12 @@ import "github.com/k0sproject/k0s/pkg/apis/k0s/v1beta1"
 // ReleaseMetadata holds the metadata about a specific release, including addons and
 // their versions.
 type ReleaseMetadata struct {
-	Versions      map[string]string
-	K0sSHA        string
-	K0sBinaryURL  string
-	K0sImages     []string
-	Configs       v1beta1.HelmExtensions
-	AirgapConfigs v1beta1.HelmExtensions
-	Protected     map[string][]string
+	Versions       map[string]string
+	K0sSHA         string
+	K0sBinaryURL   string
+	K0sImages      []string
+	Configs        v1beta1.HelmExtensions            // always applied
+	AirgapConfigs  v1beta1.HelmExtensions            // applied for airgap installations
+	BuiltinConfigs map[string]v1beta1.HelmExtensions // applied if the relevant builtin addon is enabled
```

Comment:
Is this field intended for usage with other builtin extensions as well (other than Velero) ? Let's suppose I want to override OpenEBS config, will I be able to use this field for it ?

Reply:
This is intended for any future things we add as 'builtin' items, like distributed storage when that becomes a thing